### PR TITLE
Re-organize iOS code into class-continuation categories

### PR DESCRIPTION
**Pull request by JanGorman**
## Motivation

Follow up from the discussion on a previous PR: https://github.com/react-native-image-picker/react-native-image-picker/pull/1654#discussion_r620897854

One of the edits undid my change to put the `PHPickerViewControllerDelegate` conformance into a class-continuation for reasons of consistency. Matt Galloway's book [Effective Objective-C 2.0](https://www.effectiveobjectivec.com) recommends this technique. We hide implementation detail and it helps with readability since we organise the code into units of specific functionality.

## Test Plan

Smoke test using the embedded Example app. There are no changes in terms of functionality.

### CODE REVIEW

File: ios/ImagePickerManager.m [hunk 1]
```diff
@@ -14,6 +14,12 @@ @interface ImagePickerManager ()
 
 @end
 
+@interface ImagePickerManager (UIImagePickerControllerDelegate) <UINavigationControllerDelegate, UIImagePickerControllerDelegate>
```
Comment (JanGorman) [inline ios/ImagePickerManager.m:17]: The `UIImagePickerController`'s delegate is defined as `UIImagePickerControllerDelegate & UINavigationControllerDelegate` so it makes sense to put both into a single continuation

File: ios/ImagePickerManager.h [hunk 2]
```diff
@@ -1,12 +1,11 @@
 #import <React/RCTBridgeModule.h>
 #import <UIKit/UIKit.h>
-#import <PhotosUI/PhotosUI.h>
 
 typedef NS_ENUM(NSInteger, RNImagePickerTarget) {
   camera = 1,
   library
 };
 
-@interface ImagePickerManager : NSObject <RCTBridgeModule, UINavigationControllerDelegate, UIActionSheetDelegate, UIImagePickerControllerDelegate, PHPickerViewControllerDelegate>
```
Comment (JanGorman) [inline ios/ImagePickerManager.h:10]: `UIActionSheetDelegate` conformance was declared but not required anywhere. Remove it completely

File: ios/ImagePickerManager.m [hunk 3]
```diff
@@ -156,9 +108,6 @@ - (void)onImageObtained:(UIImage*)image data:(NSData*)data
     else if ([fileType isEqualToString:@"png"]) {
         data = UIImagePNGRepresentation(image);
     }
-    else {
-        data = data;
```
Comment (JanGorman) [inline ios/ImagePickerManager.m:160]: This assignment of the parameter `data` to itself can be skipped :)

File: ios/ImagePickerManager.m [hunk 4]
```diff
@@ -82,60 +88,6 @@ - (void) showPickerViewController:(UIViewController *)picker
     });
 }
 
-- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info
```
Comment (JanGorman) [inline ios/ImagePickerManager.m:85]: Moved 1:1 [here](https://github.com/react-native-image-picker/react-native-image-picker/pull/1702/files#diff-e80fbe0919b9194e068e83807bd44a593c476cc4a9b787c284ec3283445d163cR298)

File: ios/ImagePickerManager.m [hunk 5]
```diff
@@ -82,60 +88,6 @@ - (void) showPickerViewController:(UIViewController *)picker
...
-- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker
-{
-    dispatch_async(dispatch_get_main_queue(), ^{
-        [picker dismissViewControllerAnimated:YES completion:^{
-            self.callback(@[@{@"didCancel": @YES}]);
-        }];
-    });
-}
-
-- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)){
```
Comment (JanGorman) [inline ios/ImagePickerManager.m:111]: Moved 1:1 [here](https://github.com/react-native-image-picker/react-native-image-picker/pull/1702/files#diff-e80fbe0919b9194e068e83807bd44a593c476cc4a9b787c284ec3283445d163cR328)

**Comment by Johan-dutoit** [comment]
Would this fix https://github.com/react-native-image-picker/react-native-image-picker/issues/1698?

I'm not facing the issue myself, so can't be certain.

**Comment by JanGorman** [comment]
@Johan-dutoit Let me verify, could be that it requires an additional `API_AVAILABLE` macro. I'll kick off an XCode 11 installation to verify. I'll report back asap

**Comment by JanGorman** [comment]
@Johan-dutoit I amended the PR to include some `#if __has_include` statements. I would say it's up to you as maintainer if you want to go this route or not. The project states iOS version compatibility but not a certain Xcode version compatibility. Thus, you could also make it clear there version 3.6+ requires Xcode 12+. Considering that Xcode 13 is less than a month away that might be fair

**Comment by Johan-dutoit** [comment]
Thanks @JanGorman, with all the [heat](https://github.com/facebook/react-native/issues/31480) around xcode 12, I'm hesitant to make it a minimum requirement right now.   Once that settles, I'm more than happy to introduce that requirement (which should remove the `#if __has_include` statements right?).

p.s. I'm also happy to bring you onboard as a maintainer if you're up for it, or if you'd only like to contribute when you can (as you have been), then that's okay too.

**Comment by JanGorman** [comment]
@Johan-dutoit 

> I'm more than happy to introduce that requirement (which should remove the #if __has_include statements right?).

Correct.

> I'm also happy to bring you onboard as a maintainer if you're up for it

Sure, sounds good – what are your expectations towards maintainers? I'll be a bit under water for the next few weeks as I'm in the process of moving but happy to support more once everything's settled

**Comment by Johan-dutoit** [comment]
@JanGorman some conflicts to resolve, will happily merge afterwards.

re the maintainer, it would be PR reviews and general issue support.

**Comment by JanGorman** [comment]
@Johan-dutoit Took a while, I was busy moving 😅 

The conflicts are resolved. And I'd be happy to help out as maintainer

**Review by Johan-dutoit — APPROVED** [review]

**Pull request merged**